Composer/plugins/azurePublish/src/luis.ts

```diff
@@ -0,0 +1,247 @@
+      const botFiles = await this.getFiles(workingFolder);
+      const modelFiles = botFiles.filter((name) => {
+        return name.endsWith('.lu') && this.notEmptyLuisModel(name);
+      });
+
+      // Identify the generated folder
+      const generatedFolder = path.join(workingFolder, 'ComposerDialogs/generated');
+
+      // Identify the deployment settings file
+      // const deploymentSettingsPath = path.join(workingFolder, 'appsettings.deployment.json');
+
+      // Ensure the generated folder exists
+      if (!(await fs.pathExists(generatedFolder))) {
+        await fs.mkdir(generatedFolder);
+      }
+
+      // Instantiate the LuBuild object from the LU parsing library
+      // This object is responsible for parsing the LU files and sending them to LUIS
+      const builder = new luBuild.Builder((msg) =>
+        this.logger({
+          status: BotProjectDeployLoggerType.DEPLOY_INFO,
+          message: msg,
+        })
+      );
+
+      // Pass in the list of the non-empty LU files we got above...
+      const loadResult = await builder.loadContents(
+        modelFiles,
+        language || '',
+        environment || '',
+        luisAuthoringRegion || ''
+      );
+
+      // set the default endpoint
+      if (!luisEndpoint) {
+        luisEndpoint = `https://${luisAuthoringRegion}.api.cognitive.microsoft.com`;
+      }
+
+      // if not specified, set the authoring endpoint
+      if (!luisAuthoringEndpoint) {
+        luisAuthoringEndpoint = luisEndpoint;
+      }
+
+      // Perform the Lubuild process
+      // This will create new luis apps for each of the luis models represented in the LU files
+      const buildResult = await builder.build(
+        loadResult.luContents,
+        loadResult.recognizers,
+        luisAuthoringKey,
+        luisAuthoringEndpoint,
+        name,
+        environment,
+        language,
+        false,
+        loadResult.multiRecognizers,
+        loadResult.settings
+      );
+
+      // Write the generated files to the generated folder
+      await builder.writeDialogAssets(buildResult, true, generatedFolder);
+
+      this.logger({
+        status: BotProjectDeployLoggerType.DEPLOY_INFO,
+        message: `lubuild succeed`,
+      });
+
+      // Find any files that contain the name 'luis.settings' in them
+      // These are generated by the LuBuild process and placed in the generated folder
+      // These contain dialog-to-luis app id mapping
+      const luisConfigFiles = (await this.getFiles(workingFolder)).filter((filename) =>
+        filename.includes('luis.settings')
+      );
+      const luisAppIds: any = {};
+
+      // Read in all the luis app id mappings
+      for (const luisConfigFile of luisConfigFiles) {
+        const luisSettings = await fs.readJson(luisConfigFile);
+        Object.assign(luisAppIds, luisSettings.luis);
+      }
+
+      // Create the base LUIS config object
+      // const luisConfig: any = {
```

Comment:
I will go in and clean this out - we no longer need to construct the configuration like this here.

Reply:
> > Notice Because we use daily build botbuilder-js, so please set your registry into myget before try.
> > FYI like npm config set registry https://botbuilder.myget.org/F/botbuilder-v4-js-daily/npm/
> 
> Is this a must have or can we put this in some config file?

No, because I set the registry as myget in .npmrc file in runtime/node folder. It also make sure the azure webapp service can download those packages after we deployed.